### PR TITLE
Refurbish all documentation

### PR DESCRIPTION
**Pull request by bsian03**
So atm there are a few "issues" with the documentation of Eris, mostly with TypeScript documentation:
* TypeScript projects have no intellisense documentation whatsoever, as JSDoc in the core JS files do not get translated over. This PR adds that
* There are a couple (this may change to a load as I go on) of missing documentation for properties/methods for both TS and JSDoc. This PR adds those. Includes RequestHandler and SequentialBucket which are both added in this PR
* Some of the types in the TypeScript documentation are incorrect. I've painstakingly tracked down each and every line of code to correct this.
* The formatting of TypeScript documentation is readable, but doesn't make a lot of sense. E.g. there are a few duplicates, but why are they not referenced by a single source? I've made this a little bit cleaner.
* (something I haven't done yet, and would prefer some feedback/suggestions on this) The layout of TS Documentation is all jumbled up. i.e. interfaces are all the way at the top and at the bottom, with type declarations and classes (though some are grouped together) nested in between those interfaces. This hasn't been done yet and I'll edit it when I get to it.

I plan to fix all of these issues to the best of my ability. This will take a long time however.

**This is a WIP. If you have any feedback, please let me know and I will sort them out on the fly**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { EventEmitter } from "events";
 import { Readable as ReadableStream } from "stream";
 import { Agent as HTTPSAgent } from "https";
 import { IncomingMessage } from "http";
+import WebSocket = require("ws");
```
Comment (bsian03) [inline index.d.ts]:
```suggestion
```
Only doing this because i cba stashing my uncommited changes

File: index.d.ts [hunk 2]
```diff
@@ -1946,87 +1986,304 @@ declare namespace Eris {
...
+    file: Buffer;
+    /** What to name the file */
+    name: string;
+  }
+
+  /**
+   * Handles API requests
+   */
+  class RequestHandler implements SimpleJSON {
+    agent: unknown; // HTTP Agent - Do we parse our preferred agent here?
```
Comment (bsian03) [inline index.d.ts:2167]: cc @abalabahaha need clarification on this please
This is referenced from `Client.options.agent`

**Comment by Skillz4Killz** [comment]
Small reminder as discussed on discord:

Change the @arg to @prop on the jsdocs
https://github.com/abalabahaha/eris/blob/dev/lib/rest/RequestHandler.js#L175-L185

**Comment by bsian03** [comment]
I'll come back to this once I stop starting more projects

**Pull request closed without merging**